File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> triangle = makePascals(rowIndex+1);
        return triangle[rowIndex];
    }

vector<vector<int>> makePascals(int numRows) {
        vector<vector<int>> pascals;
        for(int i=1;i<=numRows;i++){
            vector<int> row;
            if(i==1){
                row.emplace_back(1);
            }else if(i==2){
                row.emplace_back(1);
                row.emplace_back(1);
            }else{
                vector<int> prevRow = pascals[i-2];
                int l = prevRow.size();
                row.emplace_back(1);
                for(int j = 0;j<l-1;j++){
                    row.emplace_back(prevRow[j] + prevRow[j+1]);
                }
                row.emplace_back(1);
            }
            pascals.emplace_back(row);
        }
        return pascals;
    }
    
};

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        int numRows = rowIndex+1;
        int counter = 0;
        List<List<Integer>> result = new ArrayList<>();
        
        for(int i = 1;i<=numRows;i++){
            List<Integer> row = new ArrayList<>();
            if(i==1){
                row.add(1);
            }else{
                int temp = result.get(counter-1).size();
                row.add(1);
                for(int j=0;j<temp-1;j++){
                    row.add(result.get(counter-1).get(j)+result.get(counter-1).get(j+1));
                }
                row.add(1);
            }
            result.add(row);
            counter++;
        }
        return result.get(numRows-1);
    }
}